src/main/kotlin/demo/controllers/OfferController.kt

```diff
@@ -17,36 +18,44 @@ import kotlin.NoSuchElementException
 import kotlin.collections.HashSet
 
 @RestController
+@Transactional
```

Comment:
https://stackoverflow.com/questions/3164072/large-objects-may-not-be-used-in-auto-commit-mode/53520767